mlir/lib/Conversion/ArithToEmitC/ArithToEmitC.cpp

```diff
@@ -39,6 +39,54 @@ class ArithConstantOpConversionPattern
   }
 };
 
+/// Return an operation that returns true (in i1) when operand is NaN.
+emitc::CmpOp isNan(ConversionPatternRewriter &rewriter, Location loc,
+                   Value operand) {
+  // A value is NaN exactly when it compares unequal to itself.
+  return rewriter.create<emitc::CmpOp>(
+      loc, rewriter.getI1Type(), emitc::CmpPredicate::ne, operand, operand);
+}
+
+class CmpFOpConversion : public OpConversionPattern<arith::CmpFOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(arith::CmpFOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    emitc::CmpPredicate predicate;
+    switch (op.getPredicate()) {
+    case arith::CmpFPredicate::UGT:
+      // unordered or greater than
+      predicate = emitc::CmpPredicate::gt;
```

Comment:
I think this generates the wrong code right now. C++ `>` is `ordered and greater than`, so we should emit an additional check for `nan` and an emitc.or.